crates/agglayer-node/src/node.rs

```diff
@@ -69,7 +69,10 @@ impl Node {
         let signer = ConfiguredSigner::new(config.clone()).await?;
         let address = signer.address();
         // Create a new L1 RPC provider with the configured signer.
-        let rpc = Provider::<Http>::try_from(config.l1.node_url.as_str())?
+        let client = reqwest::Client::builder()
+            .timeout(std::time::Duration::from_secs(45))
```

Comment:
The settlement on L1 is waiting for `N` confirmations, maybe 45sec can be too short. The average block time on ETH is 12.06, so I think we should be good.
This also needs to be configurable to be able to update this value without rebuilding the agglayer

Reply:
> The settlement on L1 is waiting for `N` confirmations, maybe 45sec can be too short. The average block time on ETH is 12.06, so I think we should be good.

I think the checking whether we have the required number of confirmations does not happen in single RPC request, or does it? Will double check once again.

> This also needs to be configurable to be able to update this value without rebuilding the agglayer

Reply:
> > The settlement on L1 is waiting for `N` confirmations, maybe 45sec can be too short. The average block time on ETH is 12.06, so I think we should be good.
> 
> I think the checking whether we have the required number of confirmations does not happen in single RPC request, or does it? Will double check once again.

Yea, basically  `await`ing on `ethers::PendingTransaction` periodically polls `get_transaction` until it shows up in the chain. When that is successful, it periodically polls `get_transaction_receipt` until the sufficient number of confirmations is obtained. The timeout introduced above applies to the single `get_transaction` or `get_transaction_receipt` request.

However, there is a different issue. If I'm reading the code correctly, there is no limit on the number of retries of these requests. So for example, if the L1 node is down, it keeps retrying indefinitely. As far as I can see, there is a mechanism to flag it up to the agglayer node operators (other than perhaps some very low-level logging).

Reply:
Ah sorry, I though the PR was ready to review 😅 

The `retries` defined in the `PendingTransaction` is for unconfirmed transaction, I do not see any limit to the number of request too

Reply:
> Ah sorry, I though the PR was ready to review 😅

It was at the time you reviewed, so it's a mistake on my part.